### ISSUE TITLE
Refactor start_network method in sui_commands

### ISSUE THREAD

**Issue by arun-koshy** [MystenLabs/sui#552]
Based on the discussion in PR#454, Francois suggested the following 

> ...the pattern of what's going on in `start_network` and `genesis` in `sui_commands` doesn't make sense: the task waits on each launched server to finish (rather than wait for each server to start) here:
> https://github.com/MystenLabs/fastnft/blob/dba291bfac0d9a9811e9a296f3f4712744ef2df6/sui/src/sui_commands.rs#L86-L88
> As we've seen, that's not really usable in a task, which should keep a handle rather than block on completion.
> 
> Would you and @patrickkuo open an issue to track the refactoring of `start_network` in a way that:
> - has clear semantics on starting the server, 
> - makes this usable in a tokio task (by returning the handle of the `SpawnedServer` rather blocking on it,
> - de-duplicates the code between the present function and `sui_commands`

**Comment by arun-koshy** [comment]
No longer relevant after PR#1123 removed start/stop/genesis from Rest Server
